targets/simple_switch/register_access.h

```diff
@@ -0,0 +1,108 @@
+/* Copyright 2019-present Cisco Systems, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Andy Fingerhut (jafinger@cisco.com)
+ *
+ */
+
+#ifndef SIMPLE_SWITCH_REGISTER_ACCESS_H_
+#define SIMPLE_SWITCH_REGISTER_ACCESS_H_
+
+#define PACKET_LENGTH_REG_IDX             0
+
+#define CLONE_MIRROR_SESSION_ID_REG_IDX   1
+#define CLONE_MIRROR_SESSION_ID_MASK      0x000000000000ffff
+#define CLONE_MIRROR_SESSION_ID_SHIFT     0
+#define CLONE_FIELD_LIST_REG_IDX          1
+#define CLONE_FIELD_LIST_MASK             0x00000000ffff0000
+#define CLONE_FIELD_LIST_SHIFT            16
+#define LF_FIELD_LIST_REG_IDX             1
+#define LF_FIELD_LIST_MASK                0x0000ffff00000000
+#define LF_FIELD_LIST_SHIFT               32
+#define RESUBMIT_FLAG_REG_IDX             1
+#define RESUBMIT_FLAG_MASK                0xffff000000000000
+#define RESUBMIT_FLAG_SHIFT               48
+
+#define RECIRCULATE_FLAG_REG_IDX          2
+#define RECIRCULATE_FLAG_MASK             0x000000000000ffff
+#define RECIRCULATE_FLAG_SHIFT            0
+
+
+class RegisterAccess {
+ public:
+    static void clear_all(Packet &pkt) {
+        // except do not clear packet length
+        pkt.set_register(1, 0);
+        pkt.set_register(2, 0);
+    }
+    static uint16_t get_clone_mirror_session_id(Packet &pkt) {
+        uint64_t rv = pkt.get_register(CLONE_MIRROR_SESSION_ID_REG_IDX);
+        return (uint16_t) ((rv & CLONE_MIRROR_SESSION_ID_MASK) >>
```

Comment:
use `static_cast<T>` for all numerical casts

Reply:
done